src/strands/agent/conversation_manager/token_limit_recovery.py

```diff
@@ -0,0 +1,64 @@
+"""Shared utility for handling token limit recovery in conversation managers."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from ...types.content import ContentBlock, Message
+from ...types.exceptions import MaxTokensReachedException
+from ...types.tools import ToolUse
+
+if TYPE_CHECKING:
+    from ...agent.agent import Agent
+
+logger = logging.getLogger(__name__)
+
+
+def recover_from_max_tokens_reached(agent: "Agent", exception: MaxTokensReachedException) -> None:
```

Comment:
Can we export this from one of our modules - if we expect folks to want to replicate this in their own managers, then we should make it publicly available.

Unless this because the base behavior of the `CoversationManager.handle_token_limit_reached`